pre-cluster-stamp.bicep

```
@@ -0,0 +1,411 @@
+targetScope = 'resourceGroup'
+
+/*** PARAMETERS ***/
+
+@description('The regional network spoke VNet Resource ID that the cluster will be joined to')
+@minLength(79)
+param targetVnetResourceId string
+
+@allowed([
+  'australiaeast'
+  'canadacentral'
+  'centralus'
+  'eastus'
+  'eastus2'
+  'westus2'
+  'francecentral'
+  'germanywestcentral'
+  'northeurope'
+  'southafricanorth'
+  'southcentralus'
+  'uksouth'
+  'westeurope'
+  'japaneast'
+  'southeastasia'
+])
+@description('AKS Service, Node Pools, and supporting services (KeyVault, App Gateway, etc) region. This needs to be the same region as the vnet provided in these parameters.')
+@minLength(4)
+param location string = 'eastus2'
+
+@allowed([
+  'australiasoutheast'
+  'canadaeast'
+  'eastus2'
+  'westus'
+  'centralus'
+  'westcentralus'
+  'francesouth'
+  'germanynorth'
+  'westeurope'
+  'ukwest'
+  'northeurope'
+  'japanwest'
+  'southafricawest'
+  'northcentralus'
+  'eastasia'
+  'eastus'
+  'westus2'
+  'francecentral'
+  'uksouth'
+  'japaneast'
+  'southeastasia'
+])
+@description('For Azure resources that support native geo-redunancy, provide the location the redundant service will have its secondary. Should be different than the location parameter and ideally should be a paired region - https://learn.microsoft.com/azure/best-practices-availability-paired-regions. This region does not need to support availability zones.')
+@minLength(4)
+param geoRedundancyLocation string = 'centralus'
+
+@description('The base 64 encoded AKS Ingress Controller public certificate (as .crt or .cer) to be stored in Azure Key Vault as secret and referenced by Azure Application Gateway as a trusted root certificate.')
+param aksIngressControllerCertificate string
```

I know it wasn't like this before, but can you add...
```suggestion
@minLength(35)
param aksIngressControllerCertificate string
```